server/src/controllers/AuthController.js

```diff
@@ -309,4 +347,31 @@ module.exports = {
       return res.status(500).json({ error: error.message });
     }
   },
+
+	async verifyemail(req, res) {
+		const errors = validationResult(req);
+
+		if (!errors.isEmpty()) {
+			res.status(422).json({ errors: errors.array() });
+			return;
+		}
+
+		try {
+			const { token } = req.body;
+			let user = await User.findOne({
+				emailVerificationToken: token,
+			});
+
+			if (user) {
+				user.isEmailVerified = true;
+				user.emailVerificationToken = undefined;
+				await user.save();
```

Comment:
Here, we should generate the `token` and `refreshToken` and add them.

Reply:
Here, we are just verifying the email address, so there isn't a need for generating `token` and `refreshToken`

Reply:
Then as per the current implementation, there is no place where `token` and `refreshToken` can be used.

Keeping in mind that these values are first created when a user *successfully logs in*, it would make sense to add them here, wouldn't it ?

Reply:
Verify Email and Login are separate pages in themselves. So once email is verified using the verify email page, the user can move on to the login page to log in and that is where the `token` & `refresh token` is added. 

Routes :
```
  app.post('/api/login', auth.validate('login'), AuthController.login) 
  app.post('/api/verify-email', AuthController.verifyemail)
```